src/server.rs

```diff
@@ -354,6 +356,25 @@ async fn list(config: web::Data<RwLock<Config>>) -> Result<HttpResponse, Error>
                     }
                 };
                 let mut file_name = PathBuf::from(e.file_name());
+
+                let creation_date_utc = match metadata
+                    .created()
+                    .map(|v| v.duration_since(UNIX_EPOCH)
+                        .expect("Time since UNIX epoch should be valid.")
+                        .as_millis()
+                    )
+                {
+                    Ok(millis) => Some(
+                        uts2ts::uts2ts(
+                            i64::try_from(millis)
+                                .expect("UNIX time should be smaller than i64::MAX")
+                            / 1000
+                        )
+                        .as_string()
+                    ),
+                    Err(_) => None
+                };
```

Comment:
If we are going to ignore the error then this can be refactored into:

```suggestion
                let creation_date_utc = metadata
                    .created()
                    .ok()
                    .map(|v| {
                        v.duration_since(UNIX_EPOCH)
                            .expect("Time since UNIX epoch should be valid.")
                            .as_millis()
                    })
                    .map(|millis| {
                        uts2ts::uts2ts(
                            i64::try_from(millis)
                                .expect("UNIX time should be smaller than i64::MAX")
                                / 1000,
                        )
                        .as_string()
                    });
```